dotCMS/src/integration-test/java/com/dotmarketing/servlets/BinaryExporterServletTest.java

```diff
@@ -0,0 +1,209 @@
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(DataProviderRunner.class)
+public class BinaryExporterServletTest {
+
+    // Temporary binary png file
+    private static class TmpBinaryFile implements Closeable {
+
+        private final Path pngFilePath;
+
+        public TmpBinaryFile(boolean setContent) throws IOException {
+            pngFilePath = Files.createTempFile("tmp", ".png");
+            if (setContent) {
+                Files.write(pngFilePath, ShortyServletAndTitleImageTest.pngPixel);
+            }
+        }
+
+        public Path getPath() {
+            return pngFilePath;
+        }
+
+        public File getFile() {
+            return pngFilePath.toFile();
+        }
+
+        @Override
+        public void close() throws IOException {
+            Files.deleteIfExists(pngFilePath);
+        }
+    }
+
+    private static final String BY_ID = "by-identifier";
+    private static final String BY_INODE = "by-inode";
+
+    private static final String READ_PERMISSIONS = "has-read-permissions";
+    private static final String NO_PERMISSIONS = "no-permissions";
+
+    private static Host host = null;
+    private static Role role = null;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        // Set testing environment
+        IntegrationTestInitService.getInstance().init();
+
+        host = new SiteDataGen().nextPersisted();
+        role = new RoleDataGen().nextPersisted();
+    }
+
+    @DataProvider
+    public static Object[] testCases() {
+        return new String[][] {
+                { BY_ID, READ_PERMISSIONS },
+                { BY_ID, NO_PERMISSIONS },
+                { BY_INODE, READ_PERMISSIONS },
+                { BY_INODE, NO_PERMISSIONS },
+        };
+    }
+
+
+    @Test
+    @UseDataProvider("testCases")
+    public void requestBinaryFile(
+            final String byIdType, final String permissionType) throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4885343)